### ISSUE TITLE
Add weighted mean recombination rate to msprime.RecombinationMaps 

### ISSUE THREAD

**Issue by jgallowa07** [tskit-dev/msprime#728]
Happy to implement this; The only questions being whether I should do this using `numpy`, or, do total `genetic distance (Morgans) / total physical distance (bp).`

If it's the latter, I'm going to need to map physical locations in a recombination map to genetic locations.  I'm not sure if the function `physical_to_genetic` already does something similar, It looks to me like it's mapping it to the location on the scale of `self.num_loci` not Morgans.

The mapping function is pretty strait-forward in python.
```
def map_physical_to_cm(recombination_map):
    rec_map = recombination_map # self
    positions = rec_map.get_positions()
    rates = rec_map.get_rates()
    cm = [0.0]
    for i in range(1,len(positions)):
        num_bp = positions[i] - positions[i-1]
        cm.append(cm[i-1]+((rates[i-1] * num_bp) * 100))
    return cm
``` 
and adding the mean recombination rate with function this would be trivial. Any thoughts @jeromekelleher ?

**Comment by jeromekelleher** [comment]
I would use numpy. I'm planning on updating the inferface for the RecombinationMap to use numpy natively and give direct access to the arrays, so the numpy method will ultimately be quite a bit more efficient.

**Comment by jgallowa07** [comment]
closing due to #729

**Comment by jeromekelleher** [comment]
FYI, github will do this automatically if you include a line "Closes #729" in a commit message.
